apps/mobile/src/features/approver/components/asset-outcome-recipient.tsx

```diff
@@ -0,0 +1,40 @@
+import { useSip10AccountBalance } from '@/queries/balance/sip10-balance.query';
+import { useGetContractInterface } from '@/queries/stacks/contract-interface.query';
+import { deserializeAccountId } from '@/store/accounts/accounts';
+import { deserializeTransaction } from '@stacks/transactions';
+
+import { assertContractCallPayload, getContractAddress, getSip10TransferRecipient } from '../utils';
+import { OutcomeAddressesCard } from './outcome-addresses-card';
+
+export function AssetOutcomeRecipient({ txHex, accountId }: { txHex: string; accountId: string }) {
+  const tx = deserializeTransaction(txHex);
+  assertContractCallPayload(tx.payload);
+
+  const contractAddress = getContractAddress(tx.payload);
+  const { functionArgs, functionName, contractName } = tx.payload;
+
+  const { data: contractInterfaceData } = useGetContractInterface(
+    contractAddress,
+    contractName.content
+  );
+  const { fingerprint, accountIndex } = deserializeAccountId(accountId);
+  const sip10Data = useSip10AccountBalance(fingerprint, accountIndex);
+
+  if (!contractInterfaceData) return null;
+
+  const recipient = getSip10TransferRecipient({
+    functionName: functionName.content,
+    functionArgs,
+    contractInterfaceData,
+  });
+
+  if (!recipient) return null;
+
+  const token = sip10Data.value?.sip10s.find(
```

Comment:
In token details I was starting on some work to return token / balance based on `tokenId`. It could be useful to have a query / service to also accept contract ID for that so we can re-use this